src/components/ShoppingItem.re

```diff
@@ -0,0 +1,36 @@
+open Types;
+
+let component = ReasonReact.statelessComponent("ShoppingItem");
+
+let make = (~item: shoppingItem, ~onItemChange, ~id: int, _children) => {
+  let handleProductChange = event => {
```

Comment:
I think you should be able to annotate the function as follows:
`let handleProductChange = event: unit => {`
and avoid `();` at the end.

Reply:
Annotated it in `ShoppingList`.